pkg/engine/virtualmachines.go

```diff
@@ -563,3 +567,12 @@ func getVaultSecretGroup(linuxProfile *api.LinuxProfile) []compute.VaultSecretGr
 	}
 	return vaultSecretGroups
 }
+
+func addCustomTagsToVM(tags map[string]string, vm *compute.VirtualMachine) {
```

Comment:
let's add a small unit test for this

Reply:
let me know if you need any pointers on where/how

Reply:
Just did some tests, please let me know what you think about them! Thanks a lot for offering help :)